source/fe/mapping_fe.cc

```diff
@@ -2270,22 +2271,105 @@ std::vector<Point<spacedim>>
 {
-  Assert(
-    check_all_manifold_ids_identical(cell),
-    ExcMessage(
-      "All entities of a cell need to have the same manifold id as the cell has."));
+  std::vector<Point<spacedim>> mapping_support_points(
+    fe->get_unit_support_points().size());
 
   std::vector<Point<spacedim>> vertices(cell->n_vertices());
-
   for (const unsigned int i : cell->vertex_indices())
     vertices[i] = cell->vertex(i);
 
-  std::vector<Point<spacedim>> mapping_support_points(
-    fe->get_unit_support_points().size());
+  if (check_all_manifold_ids_identical(cell))
+    {
+      // version 1) all geometric entities have the same manifold:
+
+      cell->get_manifold().get_new_points(vertices,
+                                          mapping_support_point_weights,
+                                          mapping_support_points);
+    }
+  else
+    {
+      // version 1) geometric entities have different manifold
+
+      // helper function to compute mapped points on subentities
+      // note[PM]: this function currently only uses the vertices
+      // of cells to create new points on subentities; however,
+      // one should use all bounding points to create new points
+      // as in the case of MappingQ.
+      const auto process = [&](const auto    &manifold,
+                               const auto    &indices,
+                               const unsigned n_points) {
+        if ((indices.size() == 0) || (n_points == 0))
+          return;
+
+        const unsigned int n_shape_functions =
+          this->fe->reference_cell().n_vertices();
+
+        Table<2, double> mapping_support_point_weights_local(n_points,
+                                                             n_shape_functions);
+        std::vector<Point<spacedim>> mapping_support_points_local(n_points);
 
-  cell->get_manifold().get_new_points(vertices,
-                                      mapping_support_point_weights,
-                                      mapping_support_points);
+        for (unsigned int p = 0; p < n_points; ++p)
+          for (unsigned int i = 0; i < n_shape_functions; ++i)
+            mapping_support_point_weights_local(p, i) =
+              mapping_support_point_weights(
+                indices[p + (indices.size() - n_points)], i);
+
+        manifold.get_new_points(vertices,
+                                mapping_support_point_weights_local,
+                                mapping_support_points_local);
+
+        for (unsigned int p = 0; p < n_points; ++p)
+          mapping_support_points[indices[p + (indices.size() - n_points)]] =
+            mapping_support_points_local[p];
+      };
+
+      // create dummy DoFHandler to extract indices on subobjects
+      const auto                  &fe = *this->fe;
+      Triangulation<dim, spacedim> tria;
+      GridGenerator::reference_cell(tria, fe.reference_cell());
+      DoFHandler<dim, spacedim> dof_handler(tria);
+      dof_handler.distribute_dofs(fe);
```

Comment:
We really need this feature (it is one of the last things which hypercubes can do and simplices can't) but we should revise this to use a more efficient implementation. I'll have more time to look at this next week.